spec/unit/provider/service/base_spec.rb

```diff
@@ -91,4 +91,24 @@ def execute_command(command, options)
       expect {subject.stop }.to raise_error(Puppet::Error, 'Could not stop Service[test]: failed to stop')
     end
   end
+
+  context "when hasstatus is false" do
+    subject do
+      type.new(
+         :name  => "status test",
+         :provider => :base,
+         :hasstatus => false,
+         :pattern => "majestik m\u00f8\u00f8se",
+      ).provider
+    end
+
+    it "retrieves a PID from the process table" do
+      Facter.stubs(:value).with(:operatingsystem).returns("CentOS")
+      ps_output = File.binread(my_fixture("ps_ef.mixed_encoding")).force_encoding(Encoding::UTF_8)
```

Comment:
The flip side of this is what happens when the output contains valid UTF-8, but the string has been marked as something else (based on a users `LANG` + friends setup which has informed `Encoding.default_external`)?